docs/controllers/generics.md

```diff
@@ -0,0 +1,113 @@
+# Generics
+
+This chapter contains tips and tricks for re-using generic reconcilers.
+
+## Generic Reconcilers
+It is possible to create generic reconcilers by abstracing away the underlying type by using either [DynamicObject] or [PartialObjectMeta].
+
+This is a useful technique for controllers that need to do _the same thing_ to a bunch of resources, like for instance adding consistent labels / annotations.
+
+### PartialObjectMeta
+
+This is the easiest and cheapest way to create a generic reconciler because it still retains type information, and you do not have to devolve into a fully dynamic api.
+
+You can create generic reconcilers:
+
+```rust
+pub async fn reconcile<K>(obj: Arc<PartialObjectMeta<K>>, ctx: Arc<Context>)
+-> Result<Action>
+where
+    K: Resource<Scope = NamespaceResourceScope, DynamicType = ()>
+        + Clone
+        + DeserializeOwned
+        + Debug,
+{
+    let kind = K::kind(&()).to_string();
+    let ns = obj.namespace().unwrap();
+    let object_name = obj.name_any();
+    let api: Api<PartialObjectMeta<K>> = Api::namespaced(ctx.client, &ns);
+
+    // example work; apply some labels to the object
+    let patch: Patch<serde_json::Value> = get_standard_labels_for(&obj)?;
+    let serverside = PatchParams::apply("labeler");
+    api.patch(&object_name, &serverside, &patch).await?;
+
+    Ok(Action::requeue(Duration::from_secs(5 * 60)))
+}
+```
+
+The generic constraints on the associated type of the [Resource] here means this is a __namespaced__ resource (hence the unwrap). You could remove this bound (or change it for a cluster scoped only bound), but then you could not unwrap.
+
+The `DynamicType = ()` constraint is to indicate that this is one of the normal statically generated api types that we have api information for at the type level (i.e. they come from `k8s-openapi`).
+
+For information about the resource we rely on the generic [Resource] and [ResourceExt] traits which is implemented by [PartialObjectMeta].
+
+!!! note "Diverging Logic"
+
+    You will only get access to metadata of the object doing this. This can be mitigated by doing a `match` on `kind` and creating a more specific `Api<K>` inside a match arm.
+
+
+These reconcilers can be hooked up to a `Controller<K>` with another possibly generic fn.
+
+```rust
+async fn run_controller<K>(client: Client) -> Result<(), Infallible>
```

Comment:
This could just not have a return value and then it would simplify the code and explanation below (no need to mention `Infallible` and can use `tokio::join` instead).

Reply:
Yeah. The thing I was unsure about was whether it made sense to leave it there because there might be other reasons a user might want to not avoid the result type. Still, it's probably easier for people to extend something very basic. https://github.com/kube-rs/website/pull/47/commits/373847ad6ab9b0c9711d60ee8235bb86b6de9dc0